docs/tutorials/ml_inference/language_identification/ml_inference_with_language_identification_ingest.md

```diff
@@ -0,0 +1,468 @@
+
+**Note:** Replace the placeholders that start with `your_` with your own values.
+
+# Steps to connect the SageMaker model with OpenSearch
+
+## 1. Deploy Language Identification Model in SageMaker
+
+First, we'll deploy the `papluca/xlm-roberta-base-language-detection` model from Hugging Face to a SageMaker endpoint. Run the following Python code in a SageMaker Notebook.
+
+```python
+import sagemaker
+import boto3
+from sagemaker.huggingface import HuggingFaceModel
+
+try:
+    role = sagemaker.get_execution_role()
+except ValueError:
+    iam = boto3.client('iam')
+    # Replace with your SageMaker execution role name if different
+    role = iam.get_role(RoleName='sagemaker_execution_role')['Role']['Arn']
+
+# Hub Model configuration from Hugging Face
+hub = {
+    'HF_MODEL_ID':'papluca/xlm-roberta-base-language-detection',
+    'HF_TASK':'text-classification'
+}
+
+# Create Hugging Face Model Class
+huggingface_model = HuggingFaceModel(
+    transformers_version='4.37.0',
+    pytorch_version='2.1.0',
+    py_version='py310',
+    env=hub,
+    role=role,
+)
+
+# Deploy model to SageMaker Inference
+predictor = huggingface_model.deploy(
+    initial_instance_count=1,
+    instance_type='ml.m5.xlarge'
+)
+
+# After deployment, you can find your endpoint name in the
+# Amazon SageMaker > Inference > Endpoints console.
+# It will look something like: huggingface-pytorch-inference-2024-07-08-12-34-56-789
+print(f"SageMaker Endpoint Name: {predictor.endpoint_name}")
+
+# Test model prediction
+prediction = predictor.predict({
+    "inputs": "I like you. I love you",
+})
+print(prediction)
+```
+
+Copy the `SageMaker Endpoint Name` from the output. You will need it in the following steps.
+
+## 2. Create the Connector
+
+Based on your platform, there are different steps to create a connector. Regardless of the platform, the outcome will be a `connector_id` for model registration.
+
+### A. Using Amazon OpenSearch Service
+This is the recommended approach when using Amazon's managed service.
+
+#### a. Create an IAM Role for the OpenSearch Connector
+In the AWS IAM console, create a new role with the following trust relationship. This allows the OpenSearch service to assume this role.
+
+- **Custom trust policy:**
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "opensearchservice.amazonaws.com"
+            }
+        }
+    ]
+}
+```
+
+#### b. Attach Permissions
+Create and attach the following two inline policies to the role.
+
+- **Policy 1: Allow SageMaker Invocation**
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "sagemaker:InvokeEndpointAsync",
```

Comment:
This async permission is not needed. There isn't any async invokes in this tutorial.